crates/typst/src/foundations/calc.rs

```diff
@@ -908,6 +909,31 @@ pub fn quo(
     Ok(floor(dividend.apply2(divisor.v, Div::div, Div::div)))
 }
 
+/// Calculates the euclidean norm of a sequence of numbers.
+///
+/// ```example
+/// #calc.norm(1, 2, -3, 0.5)
+/// ```
+#[func]
+pub fn norm(
+    /// The callsite span.
+    span: Span,
+    /// The sequence of numbers from which to calculate the norm.
+    /// Must not be empty.
+    #[variadic]
+    values: Vec<Spanned<Num>>,
+) -> SourceResult<f64> {
+    let mut sum = 0f64;
+    if values.is_empty() {
```

Comment:
It doesn’t make sense to error on an empty parameter list; the result can be `0.0` in this case.